src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -0,0 +1,67 @@
+package ru.hogwarts.school.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import org.springframework.web.bind.annotation.*;
+import ru.hogwarts.school.entity.Faculty;
+import ru.hogwarts.school.entity.Student;
+import ru.hogwarts.school.service.FacultyService;
+
+import java.util.List;
+import java.util.Set;
+
+@RestController
+@RequestMapping("/faculty")
+public class FacultyController {
+    private final FacultyService facultyService;
+
+    public FacultyController(FacultyService facultyService) {
+        this.facultyService = facultyService;
+    }
+
+    @PostMapping
+    public Faculty postFaculty(@RequestBody Faculty faculty) {
+        return facultyService.postFaculty(faculty);
+    }
+
+    @GetMapping
+    @Operation(summary = "Get all faculties")
+    public List<Faculty> findAll() {
+        return facultyService.findAll();
+    }
+
+    @GetMapping(value = "/color", params = "color")
```

Comment:
```suggestion
    @GetMapping(params = "color")
```
снова та же ошибка - в пути ничего писать не нужно здесь

Reply:
Дима, если этого не сделать, сваггер воспримет метод GET, как единый и начнутся проблемы. Сваггер будет требовать все три параметра и age и  minAge и maxAge. Мы же это уже зафиксировали на прошлом пулл-реквесте. Ок, если это это ошибка - то каково решение?

Reply:
ну в данном случае swagger ui объединяет методы искать по цвету и искать по цвету/или/имени. и требует эти 2 параметра, т к они обязательные

Reply:
> если это это ошибка - то каково решение?

Не пользоваться Swagger UI, потому что он багнутый. То, что он объединяет, это только один из миллиона его багов. Я про это говорил на разборе каком-то вроде бы, далее на дипломе снова вляпаешься, если будешь продолжать им пользоваться. Запросы лучше делать через Postman - так HTTP протокол выучишь лучше и багов никаких там нет, потому что ты сам пишешь запрос, а не багнутый генератор тебе его сгенерит неправильно (Swagger UI).

Reply:
"Не пользоваться Swagger UI, потому что он багнутый". Надо бы с этого было начать))
"Я про это говорил на разборе каком-то вроде бы" - возможно, но я помню, ты много про него рассказывал и говорил, что это мощный инструмент для документации и т. п. А тут момент ключевой, что он не подходит для теста запросов по домашке(. Но спасибо за фидбеки, опыт в любом случае полезный! Этот канал связи (комменты по домашке) считаю очень важным для обучения и надеюсь, тебя это не обременяет)

Reply:
Вот где-то там я и должен был сказать, что возможность отправки запросов - не самая важная вещь) возможно, забыл сказать, что он багнутый, но про постман вроде говорил, что лучше им пользоваться

Reply:
Ок, сейчас поправлю аннотации и погоняю приложение на постмане!